tests/test_shadow_fork.py

```
@@ -77,26 +77,42 @@ def test_create_fails_if_base_branch_missing(self, tmp_path):
         fork = ShadowFork(repo_root=tmp_path)
 
         with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, "", ""),  # Check for uncommitted changes
-                (False, "", "fatal"),  # Base branch doesn't exist
-            ]
 
-            with pytest.raises(RuntimeError, match="does not exist"):
+            def git_side_effect(cmd, **kwargs):
```

Ruff (0.15.4)

[warning] Missing return type annotation for private function `git_side_effect`

(ANN202)